pkg/cloud/vsphere/services/interfaces.go

```diff
@@ -30,3 +30,14 @@ type VirtualMachineService interface {
 	// DestroyVM powers off and removes a VM from the inventory.
 	DestroyVM(ctx *context.MachineContext) (infrav1.VirtualMachine, error)
 }
+
+// LoadBalancerService is service that reconciliate load balancers
+type LoadBalancerService interface {
+
+	// Reconcile reconciles a load balancer for vsphereCluster using controlPlaneEndpoints
+	// to serve as api endpoint
+	Reconcile(vsphereCluster *infrav1.VSphereCluster, controlPlaneIPs []string) (infrav1.APIEndpoint, error)
```

Comment:
Hi @yastij,

Shouldn't the first parameter be a `CAPI` cluster? Otherwise how is the service generic across infra providers?